wordpress-mobile/WordPress-iOS
# Issues/3786 post list sorting

Refs #3786
* Requires #3805 *

This PR builds on work done in #3805.  The relevant changes begin in dd7fa34. 

The goal is to add two new helper or "meta" properties to help with sorting posts and pages returned from an NSFetchRequest. The new properties are automatically updated in response to changes to a posts date created or its remote status. 

To ensure any existing posts get the correct values set, this PR includes a core data revision, mapping model, and migration policy for posts and pages. 

### TESTING
Setup:  Prepare a local only post in the simulator *BEFORE* switching to this branch. For convenience, be sure to use a blog with very view posts  Follow the steps outlined in this video to create a local-only post https://cloudup.com/iOZ-BQXHqyx 

Test 1: 
- Launch the simulator. Confirm there are no errors launching the app and you do not have to sign in again. 
- View the post list.  The local only post should be first in the list. 

Test 2: 
- Use a [link conditioner](http://nshipster.com/network-link-conditioner/) to reduce your network speed.  A slow connection will help you verify this is working. 
- Edit an existing post. Its easier if its near the top of the list, but not THE first one in the list. 
- Set it to publish immediately and save your changes. 
- Watch closely and confirm that it is the first post in the list when publishing. 

Needs Review: @koke @astralbodies
